libevmasm/Assembly.cpp

```diff
@@ -853,20 +887,117 @@ std::map<u256, u256> const& Assembly::optimiseInternal(
 	return *m_tagReplacements;
 }
 
+namespace
+{
+template<typename ValueT>
+void setBigEndian(bytes& _dest, size_t _offset, size_t _size, ValueT _value)
+{
+	assertThrow(numberEncodingSize(_value) <= _size, AssemblyException, "");
+	toBigEndian(_value, bytesRef(_dest.data() + _offset, _size));
+}
+
+template<typename ValueT>
+void appendBigEndian(bytes& _dest, size_t _size, ValueT _value)
+{
+	_dest.resize(_dest.size() + _size);
+	setBigEndian(_dest, _dest.size() - _size, _size, _value);
+}
+
+template<typename ValueT>
+void setBigEndianUint16(bytes& _dest, size_t _offset, ValueT _value)
+{
+	setBigEndian(_dest, _offset, 2, _value);
+}
+
+template<typename ValueT>
+void appendBigEndianUint16(bytes& _dest, ValueT _value)
+{
+	static_assert(!std::numeric_limits<ValueT>::is_signed, "only unsigned types or bigint supported");
+	assertThrow(_value <= 0xFFFF, AssemblyException, "");
+	appendBigEndian(_dest, 2, static_cast<size_t>(_value));
+}
+}
+
+/// EOF header bytecode | code section sizes offsets | data section size offset
+std::tuple<bytes, std::vector<size_t>, size_t> Assembly::createEOFHeader(std::set<uint8_t> const& referencedSubIds) const
```

Comment:
I think you forgot to change the signature in the header file. Might be why CI is failing.

Also, missing `_`:
```suggestion
std::tuple<bytes, std::vector<size_t>, size_t> Assembly::createEOFHeader(std::set<uint8_t> const& _referencedSubIds) const
```

Reply:
Oh, and by the way, I think that a better place for `///` docstrings is in the header.

Reply:
Ah, wait. It's there. It's just that `appendEOFHeader()` has not been removed from there and is unused :) So it should be removed but can't be the reason for the failure.

Reply:
Yeah. Removed. CI failed because some other reason. Should be ok now.